src/java/org/apache/cassandra/db/compaction/SharedCompactionObserver.java

```diff
@@ -55,7 +77,14 @@ public void registerExpectedSubtask()
     public void onInProgress(CompactionProgress progress)
     {
         if (inProgressReported.compareAndSet(null, progress))
-            observer.onInProgress(progress);
+        {
+            Throwable err = null;
+            for (CompactionObserver compObserver : compObservers)
+                err = Throwables.perform(err, () -> compObserver.onInProgress(progress));
+
+            if (err != null)
+                logger.error("Failed to notify CompactionObserver.onInProgress for {}", progress.operationId(), err);
```

Comment:
Proceeding after an error here is not safe (in theory this could mean a broken index or something worse).

Do `Throwables.maybeFail(error)` so that the compaction aborts.

Reply:
fixed in https://github.com/datastax/cassandra/pull/1774/commits/6cf01dd6a3d23bb7c5a9ef787b4b6e49f5ed6689